docs/cluster_and_device_type_dev/how_to_add_new_dts_and_clusters.md

```diff
@@ -0,0 +1,248 @@
+            2. In `ApplicationShutdown()`, add a call to your
+               `Clusters::XYZMode::Shutdown();`.
+        6. In
+           [examples/all-clusters-app/linux/BUILD.gn](https://github.com/project-chip/connectedhomeip/tree/master/examples/all-clusters-app/linux/BUILD.gn),
+           add the `xyz-mode.cpp` file to the `sources` section.
+        7. In
+           [src/app/common/templates/config-data.yaml](https://github.com/project-chip/connectedhomeip/blob/master/src/app/common/templates/config-data.yaml):
+            1. Add a `<XYZMode>::ModeTag` to the `EnumsNotUsedAsTypeInXML`
+               section.
+            2. Add an `XYZ Mode` entry to the
+               `CommandHandlerInterfaceOnlyClusters` section.
+        8. In
+           [src/app/util/util.cpp](https://github.com/project-chip/connectedhomeip/blob/master/src/app/util/util.cpp),
+           in the `// Cluster Init Functions...` section, add a void
+           `MatterXYZModePluginServerInitCallback() {}` definition.
+        9. In
+           [src/app/zap-templates/zcl/zcl-with-test-extensions.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap-templates/zcl/zcl-with-test-extensions.json):
+            1. Add the `xyz-mode-cluster.xml` to the `xmlFile` list
+            2. In the `attributeAccessInterfaceAttributes` entry, add your new
+               entry
+               `"XYZ Mode": [ "SupportedModes", "CurrentMode", "FeatureMap" ]` -
+               this will mean ZAP won't generate code to handle these attributes
+        10. In
+            [src/app/zap_cluster_list.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/zap_cluster_list.json):
+            1. Add your `XYZ_MODE_CLUSTER: []` to `ClientDirectories: { }`
+               object
+            2. Add your `"XYZ_MODE_CLUSTER": ["mode-base-server"]` to
+               `"ServerDirectories": { }`
+    4. The code under
+       [src/app/tests/suites/certification](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/certification)
+       for YAML or
+       [src/python_testing](https://github.com/project-chip/connectedhomeip/tree/master/src/python_testing)
+       for Python should ideally implement the test plan (section 4 below).
+    5. A test under
+       [src/app/tests/suites](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites)
+       (not certification) can test anything, in addition to, or preceding the
+       official YAML representing the official test plan.
+4. Add the test plan, using the templates below:
+
+    1. [cluster_test_plan_template.adoc](https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/template/cluster_test_plan_template.adoc)
+    2. [test_plan_template.adoc](https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/template/test_plan_template.adoc)
+
+    Also, join this private Slack channel: csg-tt-test-plans
```

Comment:
```suggestion
    Also, join this private Slack channel: `csg-tt-test-plans`
```

Reply:
It is not as clear how to join a private slack channel. Maybe instruct users to ask others (where?) to add them to this channel?